service/src/main/java/uk/gov/hmcts/reform/fpl/service/ApplicantMigrationService.java

```diff
@@ -0,0 +1,99 @@
+package uk.gov.hmcts.reform.fpl.service;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.collect.ImmutableMap;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.ccd.client.model.AboutToStartOrSubmitCallbackResponse;
+import uk.gov.hmcts.reform.ccd.client.model.CaseDetails;
+import uk.gov.hmcts.reform.fpl.enums.PartyType;
+import uk.gov.hmcts.reform.fpl.model.ApplicantParty;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+
+import static java.util.stream.Collectors.toList;
+
+@Service
+public class ApplicantMigrationService {
+
+    @Autowired
+    private final ObjectMapper mapper = new ObjectMapper();
+
+    public AboutToStartOrSubmitCallbackResponse setMigratedValue(CaseDetails caseDetails) {
+        Map<String, Object> data = caseDetails.getData();
+
+        if (caseDetails.getData().containsKey("applicants") || !caseDetails.getData().containsKey("applicant")) {
+            data.put("applicantsMigrated", "Yes");
+
+            if (!caseDetails.getData().containsKey("applicants")) {
+                List<Map<String, Object>> populatedApplicant = new ArrayList<>();
+                populatedApplicant.add(ImmutableMap.of(
+                    "id", UUID.randomUUID().toString(),
+                    "value", ImmutableMap.of(
+                        "party", ImmutableMap.of(
+                            "partyId", UUID.randomUUID().toString()
```

Comment:
could you add a comment as to why we are setting partyId here

Reply:
👍